ndmg/utils/gen_utils.py

```diff
@@ -421,6 +420,17 @@ def inner(*args, **kwargs):
 
     return outer
 
+def timer(f):
+    """Print the runtime of the decorated function"""
+    @functools.wraps(f)
+    def wrapper_timer(*args, **kwargs):
+        start_time = time.perf_counter()    # 1
+        a = f(*args, **kwargs)
+        end_time = time.perf_counter()      # 2
+        run_time = end_time - start_time    # 3
+        print(f"Function {f.__name__!r} finished in {run_time:.4f} secs")
```

Comment:
oh, cool, I actually had no idea that you could do `!r` and `:.4f` inside the `{}` in f-strings.
I understand why you're doing the `.4f` (because we only want 4 decimal places),
but why are you using the function's `__repr__` instead of its `__str__` with `!r`?